clickhouse_driver/client.py

```diff
@@ -645,12 +645,23 @@ def send_data(self, sample_block, data, types_check=False, columnar=False):
         else:
             slicer = column_chunks if columnar else chunks
 
+        revision = self.connection.server_info.used_revision
+        should_wait_for_profile_event = (
+            revision >=
+            defines.DBMS_MIN_PROTOCOL_VERSION_WITH_PROFILE_EVENTS_IN_INSERT
+        )
+
         for chunk in slicer(data, client_settings['insert_block_size']):
             block = block_cls(sample_block.columns_with_types, chunk,
                               types_check=types_check)
             self.connection.send_data(block)
             inserted_rows += block.num_rows
 
+            # Starting from the specific revision there are profile events
+            # sent by server in response to each inserted block
+            if should_wait_for_profile_event:
+                self.receive_packet()
```

Comment:
I think it is better to explicitly allow only `ServerPacketTypes.PROFILE_EVENTS` (to make it compatible with ClickHouse code base).

Reply:
I've used the existing function cause it receives the packet and also correctly handles the exception type. And as I get from sources, there is no extra processing function for ProfileEvent. 
My goal was a minimal intervention, cause I am not familiar enough with overall packet processing in the library.

Reply:
As I get it you propose adding some method like this on connection class?
```python
def receive_profile_event(self):
    packet_type = read_varint(self.fin)
    if packet_type == ServerPacketTypes.PROFILE_EVENTS:
        self.receive_data(may_be_compressed=False)
        return None
    elif packet_type == ServerPacketTypes.EXCEPTION:
        raise self.receive_exception()
    else:
        message = self.unexpected_packet_message('Hello or Exception',
                                                                                         packet_type)
        self.disconnect()
        raise errors.UnexpectedPacketFromServerError(message)
````

Reply:
>As I get it you propose adding some method like this on connection class?

Exactly.

Reply:
Done